shell.go

```diff
@@ -84,6 +92,94 @@ func NewShellWithClient(url string, c *gohttp.Client) *Shell {
 	}
 }
 
+// Load all shell urls with error checking
+func NewShellWithCheck(urls ...string) (*Shell, error) {
```

Comment:
Maybe `TryNewShell`? Not happy with the name but can't think of a better one.

Anyways, this should probably have some more documentation (the description doesn't explain what "error checking" means and would otherwise lead me to believe that it returns a slice of shells).